config/checkstyle_checks.xml

```diff
@@ -490,6 +490,36 @@
       <property name="allowNoEmptyLineBetweenFields" value="true"/>
       <property name="allowMultipleEmptyLinesInsideClassMembers" value="false"/>
     </module>
+    <module name="EmptyLineWrappingInBlock">
```

Comment:
Right now, only 26 classes have en empty line at the bottom:
https://pbludov.github.io/issue-3923/bottom-noel/checkstyle-with-excludes/index.html

This will require the refactoring of the whole project. For the top separator, the stats are 50/50 for "empty-line" vs "no-empty-line".

Also, you need to think about nested classes. Should they have be wrapped with empty lines?
The Orekit project uses this style:

```
class Foo {

int wrappedWithEmptyLines;

class Inner {
    int looksMoreLikeAMethod;
  }

}
```

Reply:
>  top and bottom empty lines for Class/Interface

I am fine with this even though this is not usually my preference.

> "no-empty-line" for IF, While, .....

This is ok.

> "EMPTY_LINE_ALLOWED" - for the rest - method,ladba,.....

I disagree with this. We should enforce some type of style for ALL tokens. There are too many cases where people do one way or the other, either on purpose or accident. I believe I have seen this for lambdas.
My suggestion is "no-empty-line" for the 'rest' to keep code condensed.
IMO, If code looks too packed together, it is because user isn't putting any blank line space between instructions and are crowding them together.

Reply:
>I am fine with this even though this is not usually my preference.

This the only place for empty line that I am ok with, even I it is not my best preference too. But it is ok.

>I disagree with this. We should enforce some type of style for ALL tokens. 

Ok, just tried to show usage of this option to users. While smart logic that could define whether method signature is interfering with code in body

>IMO, If code looks too packed together, it is because user isn't putting any blank line space between instructions and are crowding them together.

Multiline Method signature and condensed first line is problem, see examples in mailist. But this is off topic discussion.

Reply:
> Should they have be wrapped with empty lines?

For now let's keep them same as top classes. If check become popular users will request new property.  Inner class should follow top class format, as it could be easily move out to become top.

Reply:
>(different configs for different projects)

Please generate few reports over checkstyle code to let us be more confident on what tokens to validate by what options.